src/types/typeobject.rs

```diff
@@ -53,12 +54,55 @@ impl PyType {
         Ok(result == 1)
     }
 
+    /// Checks whether `self` is a subclass of `other`.
+    ///
+    /// Equivalent to the Python expression `issubclass(self, other)`.
+    pub fn is_subclass_of(&self, other: &PyType) -> PyResult<bool> {
+        let result = unsafe { ffi::PyObject_IsSubclass(self.as_ptr(), other.as_ptr()) };
+        err::error_on_minusone(self.py(), result)?;
+        Ok(result == 1)
+    }
+
     /// Check whether `obj` is an instance of `self`.
     ///
-    /// Equivalent to Python's `isinstance` function.
+    /// Equivalent to the Python expression `isinstance(obj, self)`.
     pub fn is_instance<T: AsPyPointer>(&self, obj: &T) -> PyResult<bool> {
```

Comment:
Should this also have `is_instance_of`? Ack that `PyAny` already has that

Reply:
If we kept the names consistent, this would be `is_instance_of`. `is_instance` is also possible, it would be a static method.

Reply:
I wonder if we should just throw this method away entirely? `obj.is_instance(other)` and `type.is_subclass(base)` work in the same order `isinstance(obj, other)`, whereas this `type.is_instance(obj)` method is the reversed `isinstance(obj, type)`.

Seems like a great way to introduce confusion and bugs, so I'd be up for deprecating (can point users to `PyAny::is_instance`).

Reply:
Oh yeah, that's a good point...In Python, you're meant to use `issubclass()` for types, `isinstance()` for objects. So I agree with deprecating this method on `PyType` - it's too confusing of sugar for something you already can achieve.